src/include/migraphx/apply_alpha_beta.hpp

```diff
@@ -0,0 +1,74 @@
+#ifndef MIGRAPHX_GUARD_MIGRAPHX_APPLY_ALPHA_BETA_HPP
+#define MIGRAPHX_GUARD_MIGRAPHX_APPLY_ALPHA_BETA_HPP
+
+#include "migraphx/errors.hpp"
+#include <migraphx/instruction.hpp>
+#include <migraphx/instruction_ref.hpp>
+#include <migraphx/make_op.hpp>
+#include <migraphx/common.hpp>
+#include <migraphx/module.hpp>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+template <typename T = float>
+instruction_ref insert_apply_alpha_beta(module& m,
+                                        instruction_ref pos,
+                                        const std::vector<instruction_ref>& args,
+                                        const std::string& op_name,
+                                        T alpha = 1.0f,
+                                        T beta  = 0.0f)
+{
+    auto arg0     = args[0];
+    auto arg1     = args[1];
+    auto dot_type = arg0->get_shape().type();
+    if(!float_equal(alpha, 1.0f))
+    {
+        auto alpha_literal = m.add_literal(alpha);
+        arg0 = insert_common_op(m, pos, migraphx::make_op("mul"), {alpha_literal, arg0});
+        if(arg0->get_shape().type() != dot_type)
+        {
+            arg0 = m.insert_instruction(pos, make_op("convert", {{"target_type", dot_type}}), arg0);
+        }
+    }
+    auto dot_res = m.insert_instruction(pos, migraphx::make_op(op_name), arg0, arg1);
+    if(args.size() == 3)
+    {
+        if(not float_equal(beta, 0.0f) && args[2]->get_shape().elements() > 0)
+        {
+            auto out_lens   = arg0->get_shape().lens();
+            out_lens.back() = arg1->get_shape().lens().back();
+            auto arg2       = args[2];
+            auto arg2_lens  = arg2->get_shape().lens();
+            dot_type        = arg2->get_shape().type();
+            if(!std::equal(out_lens.begin(), out_lens.end(), arg2_lens.begin(), arg2_lens.end()))
```

Comment:
This can be written as `out_lens != arg2_lens`. No need to use `std::equal` here.

Reply:
Done